### PR TITLE
Add email address validation for user

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: core/lib/spree/core/validators/email.rb [hunk 1]
```diff
@@ -19,6 +19,13 @@ class EmailValidator < ActiveModel::EachValidator
   HOSTNAME_PATTERN =
     /\A(?:[-\p{L}\d]{1,63}\.)+#{Regexp.union(*TLDS)}\z/i.freeze
 
+  # AR Compatible validation DSL
+  VALIDATION = IceNine.deep_freeze(
+    presence:     true,
```
Comment (dkubb) [inline core/lib/spree/core/validators/email.rb]: There are extra spaces between the key and value here.

File: backend/spec/features/admin/users_spec.rb [hunk 2]
```diff
@@ -100,12 +100,12 @@
...
 
     it 'can edit the user email' do
-      fill_in 'user_email', with: 'a@example.com99'
+      fill_in 'user_email', with: 'c@example.com'
       click_button 'Update'
 
-      expect(user_a.reload.email).to eq 'a@example.com99'
       expect(page).to have_text 'Account updated'
-      expect(find_field('user_email').value).to eq 'a@example.com99'
+      expect(user_a.reload.email).to eq 'c@example.com'
```
Comment (dkubb) [inline backend/spec/features/admin/users_spec.rb]: This should use `eql(...)`.

File: backend/spec/features/admin/users_spec.rb [hunk 3]
```diff
@@ -100,12 +100,12 @@
...
+      expect(find_field('user_email').value).to eq 'c@example.com'
```
Comment (dkubb) [inline backend/spec/features/admin/users_spec.rb]: This should use `eql(...)`.

**Comment by mbj** [comment]
@dkubb This is ready for the next pass.

**Pull request merged**